homeassistant/components/climate/__init__.py

```diff
@@ -571,9 +571,7 @@ def _convert_for_display(self, temp):
                                     self.unit_of_measurement)
 
         if self.unit_of_measurement is TEMP_CELSIUS:
-            decimal_count = 1
+            return round(value * 2) / 2
```

Comment:
The reason for this change was to show Celsius temperatures rounded to a precision 0.5, otherwise it shows a temperature that does not match what is in the ecobee ui, or thermostat.  
eg, by reverting the change, I see:
Heat 19.4 - 21.1 °C
Currently: 21.9 °C